pdr_backend/sim/sim_engine.py

```diff
@@ -271,6 +291,51 @@ def save_state(self, i: int, N: int):
         return True, False
 
     @enforce_types
+    def _get_prediction_dataset(
```

Comment:
could you rename this and the test to `_get_predictions_signals_data`? This way is more specific